### PR TITLE
Adds E2E Tests for wallet creation commands

### PR DESCRIPTION
**Pull request by ibraheem-abe**
- We are testing these commands
1. btcli wallet create
2. btcli wallet new_coldkey
3. btcli wallet new_hotkey
4. btcli wallet list 

- In each test, we verify the wallet is displayed along with the key in correct format through the cli
- We also test the physical existence of wallets in the system (including coldkeys and hotkeys)

### CODE REVIEW

File: tests/e2e_tests/subcommands/wallet/test_wallet_creations.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+Verify commands:
+* btcli w list
+* btcli w create
+* btcli w new_coldkey
+* btcli w new_hotkey
+"""
+
+
+# This function physically checks the existence of wallets (coldkeys and hotkeys)
+def verify_wallet_dir(base_path, wallet_name, hotkey_name=None):
```
Comment (basfroman) [inline tests/e2e_tests/subcommands/wallet/test_wallet_creations.py:24]: pls add at least single line docstring

File: tests/e2e_tests/subcommands/wallet/test_wallet_creations.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        if not os.path.isfile(hotkey_file_path):
+            return (
+                False,
+                f"Hotkey file {hotkey_name} not found in {wallet_name}/hotkeys",
+            )
+
+    return True, f"Wallet {wallet_name} verified successfully"
+
+
+def verify_key_pattern(output, wallet_name):
```
Comment (basfroman) [inline tests/e2e_tests/subcommands/wallet/test_wallet_creations.py:65]: pls add at least single line docstring

File: tests/e2e_tests/subcommands/wallet/test_wallet_creations.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    assert match.group(1).startswith("5"), f"{wallet_name} should start with '5'"
+
+    # Assert length of key is 48 characters
+    assert (
+        len(match.group(1)) == 48
+    ), f"Key for {wallet_name} should be 48 characters long"
+    return match.group(1)
+
+
+def test_wallet_creations(local_chain: subtensor, capsys):
```
Comment (basfroman) [inline tests/e2e_tests/subcommands/wallet/test_wallet_creations.py]: pls add at least single line docstring

**Review by basfroman — APPROVED** [review]
Looks pretty well
Just left couple of comments

File: tests/e2e_tests/subcommands/wallet/test_wallet_creations.py (hunk 1 above)
Comment (ibraheem-abe) [reply tests/e2e_tests/subcommands/wallet/test_wallet_creations.py:24]: Done!

**Pull request merged**